sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/PropagatorProvider.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.spi;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+
+/**
+ * A service provider interface (SPI) for providing additional propagators to the autoconfigured
+ * SDK. The {@link TextMapPropagator} returned from any found {@link PropagatorProvider} will be
+ * available as part of {@link OpenTelemetry#getPropagators()}. This should generally only be used
+ * for custom propagators - for any propagator supported by OpenTelemetry, it is recommended to use
+ * the {@code otel.propagators} property instead.
```

Comment:
How will this interact with `otel.propagators`? Are propagators provided by this SPI additive, or will they replace whatever is specified by the system property/env var?  I see from the implementation that it is additive, but we should absolutely document that here, for maximum clarity.

Reply:
Yeah makes sense, added a note.